src/utils/event-dispatcher.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-unneeded-ternary */
+const helpers = require('../utils/helpers');
+
+// Create `CustomEvent` with fallback
+const createCustomEvent = (eventName, detail) => {
+  try {
+    return new window.CustomEvent(eventName, { detail });
+  } catch (e) {
+    const evt = document.createEvent('CustomEvent');
+
+    evt.initCustomEvent(eventName, false, false, detail);
+
+    return evt;
+  }
+};
+
+class EventDispatcher {
+  constructor(options) {
+    this.events = [];
+    this.enabled = (options && options.enabled === false)
+      ? false
+      : true; // Defaults to 'true'
+    this.waitForBeacon = (options && options.waitForBeacon === false)
```

Comment:
Would `this.waitForBeacon = !!options.waitForBeacon` have the same effect?

Reply:
following the other comment i made, this will always wait for the beacon unless specified to be false, so i think we'd want to keep it how steve currently has it